server/src/__tests__/server.test.ts

```diff
@@ -1740,7 +1740,7 @@ describe('server', () => {
     describe('Edge or not covered cases', () => {
       it('only includes variables typed as variable_name', async () => {
         const iRanges = await getFirstChangeRanges(getRenameRequestResult(106, 4))
-        // This should be 6 if all instances within let and arithmetic
+        // This should be 6 if all instances within let and c-style for loop
```

Comment:
As I note in the commit message,

The anazlyer is now detecting variables within arithmetic expressions,
but still not let or c style for loops

So a rename on

```
for i in 1 2 3; do
	echo "$i"
done

let i=0
for (( ; i < 10; i++)); do
	echo "$((2 * i))"
done
```

Would be changed to

```
for new_name in 1 2 3; do
	echo "$new_name"
done

let i=0
for (( ; i < 10; i++)); do
	echo "$((2 * new_name))"
done
```

Which is definitely wrong, but I'm interested to hear opinions from the maintainers on this

Reply:
@mcecode what do you think about this?

Reply:
I think this is to be expected since, as far as I can tell, tree-sitter-bash only updated the types of variables from `word` to `variable_name` within arithmetic expressions. For those other constructs they're still typed as `word`. The current implementation of rename symbol only renames variables typed as `variable_name`.